metadata-io/src/test/java/com/linkedin/metadata/graph/GraphServiceTestBase.java

```diff
@@ -1583,4 +1583,45 @@ public void run() {
       assertEquals(throwables.size(), 0);
   }
 
+  @Test
+  public void testPopulatedGraphServiceGetLineageMultihop() throws Exception {
+      GraphService service = getLineagePopulatedGraphService();
+
+      EntityLineageResult upstreamLineage = service.getLineage(datasetOneUrn, LineageDirection.UPSTREAM, 0, 1000, 2);
+      assertEquals(upstreamLineage.getTotal().intValue(), 0);
+      assertEquals(upstreamLineage.getRelationships().size(), 0);
+
+      EntityLineageResult downstreamLineage = service.getLineage(datasetOneUrn, LineageDirection.DOWNSTREAM, 0, 1000, 2);
+
+      assertEquals(downstreamLineage.getTotal().intValue(), 5);
+      assertEquals(downstreamLineage.getRelationships().size(), 5);
+      Map<Urn, LineageRelationship> relationships = downstreamLineage.getRelationships().stream().collect(Collectors.toMap(LineageRelationship::getEntity,
+              Function.identity()));
+      assertTrue(relationships.containsKey(datasetTwoUrn));
+      assertEquals(relationships.get(datasetTwoUrn).getDegree().intValue(), 1);
+      assertTrue(relationships.containsKey(datasetThreeUrn));
+      assertEquals(relationships.get(datasetThreeUrn).getDegree().intValue(), 2);
+      assertTrue(relationships.containsKey(datasetFourUrn));
+      assertEquals(relationships.get(datasetFourUrn).getDegree().intValue(), 2);
+      assertTrue(relationships.containsKey(dataJobOneUrn));
+      assertEquals(relationships.get(dataJobOneUrn).getDegree().intValue(), 1);
+      assertTrue(relationships.containsKey(dataJobTwoUrn));
+      assertEquals(relationships.get(dataJobTwoUrn).getDegree().intValue(), 1);
+
+      upstreamLineage = service.getLineage(datasetThreeUrn, LineageDirection.UPSTREAM, 0, 1000, 2);
+      assertEquals(upstreamLineage.getTotal().intValue(), 3);
+      assertEquals(upstreamLineage.getRelationships().size(), 3);
+      relationships = upstreamLineage.getRelationships().stream().collect(Collectors.toMap(LineageRelationship::getEntity,
+              Function.identity()));
+      assertTrue(relationships.containsKey(datasetOneUrn));
+      assertEquals(relationships.get(datasetOneUrn).getDegree().intValue(), 2);
+      assertTrue(relationships.containsKey(datasetTwoUrn));
+      assertEquals(relationships.get(datasetTwoUrn).getDegree().intValue(), 1);
+      assertTrue(relationships.containsKey(dataJobOneUrn));
+      assertEquals(relationships.get(dataJobOneUrn).getDegree().intValue(), 1);
+
+      downstreamLineage = service.getLineage(datasetThreeUrn, LineageDirection.DOWNSTREAM, 0, 1000, 2);
+      assertEquals(downstreamLineage.getTotal().intValue(), 0);
+      assertEquals(downstreamLineage.getRelationships().size(), 0);
```

Comment:
Nice test.

Any cases for when there are _no_ downstream or upstream edges for the source urn?